easybuild/easyblocks/generic/mesonninja.py

```diff
@@ -85,12 +90,14 @@ def configure_step(self, cmd_prefix=''):
                 configure_cmd == DEFAULT_CONFIGURE_CMD):
             configure_cmd += ' setup'
 
+        builddir = self.cfg.get('builddir') or self.start_dir
+
         cmd = "%(preconfigopts)s %(configure_cmd)s --prefix %(installdir)s %(configopts)s %(sourcedir)s" % {
             'configopts': self.cfg['configopts'],
             'configure_cmd': configure_cmd,
             'installdir': self.installdir,
             'preconfigopts': self.cfg['preconfigopts'],
-            'sourcedir': self.start_dir,
+            'sourcedir': builddir,
```

Comment:
Let's fix the `sourcedir` misnomer here too, since it's internal to this easyblock anyway:

```suggestion
        cmd = "%(preconfigopts)s %(configure_cmd)s --prefix %(installdir)s %(configopts)s %(builddir)s" % {
            'configopts': self.cfg['configopts'],
            'configure_cmd': configure_cmd,
            'installdir': self.installdir,
            'preconfigopts': self.cfg['preconfigopts'],
            'builddir': build_dir,

Reply:
same comment, why `source_dir` but `installdir` ?

Reply:
I consider the `installdir` a historical mistake, `source_dir` is way more readable imho...